code/modules/paperwork/pen.dm

```diff
@@ -166,22 +166,25 @@
 			var/olddesc = O.desc
 			if(QDELETED(O) || !user.canUseTopic(O, BE_CLOSE))
 				return
-			if(olddesc == input || input == "")
+			if(input == olddesc || !input)
 				to_chat(user, "<span class='notice'>You decide against changing [O]'s description.</span>")
 			else
-				O.desc = input
+				O.AddComponent(/datum/component/rename, O.name, input)
 				to_chat(user, "<span class='notice'>You have successfully changed [O]'s description.</span>")
 				O.renamedByPlayer = TRUE
 
 		if(penchoice == "Reset")
 			if(QDELETED(O) || !user.canUseTopic(O, BE_CLOSE))
 				return
-			O.desc = initial(O.desc)
-			O.name = initial(O.name)
+
+			qdel(O.GetComponent(/datum/component/rename))
```

Comment:
```suggestion
			var/datum/component/rename/rename_component = O.GetComponent(/datum/component/rename)
			if(rename_component)
				qdel(rename_component)
```

Reply:
Why? `qdel` works fine with null.

Reply:
Eh, w/e

Reply:
simple and clean is the way that you're making me feel tonight